### PR TITLE
REFACTOR: move to useMutation for update channel notification API

### LINKED ISSUE #112: useMutation - Update channel notification (PATCH)

### PR DESCRIPTION
**Pull request by madjiebimaa**
Closes #112 

## Description
Find a list of APIs that use the method PUT/PATCH/POST and have additional conditions (success, error, etc) or need some results (isLoading, etc). APIs that have those criteria will be added a layer before accessing those APIs directly with useMutation.

## Current Tasks

APIs:
- [ ] Check the slug owner (POST)
- [ ] Update user (PATCH)
- [ ] Send question (POST)
- [ ] Add new custom og (POST)
- [ ] Update custom og (PATCH)
- [ ] Add new channel notification (POST)
- [x] Update channel notification (PATCH)

### CODE REVIEW

**Comment by mazipan** [comment]
If it's too big, let's make it smaller by merging one by one.

Wdyt? @madjiebimaa 

I will create separate issue for each API, so you can create separate PR based on that

**Comment by madjiebimaa** [comment]
> If it's too big, let's make it smaller by merging one by one.
> 
> Wdyt? @madjiebimaa
> 
> I will create separate issue for each API, so you can create separate PR based on that

so rather than having several commits in one pull request, it would be good things if I create one commit for one pull request?

File: src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx [hunk 1]
```diff
@@ -56,6 +54,7 @@ export default function SettingTelegram({
 }) {
   const { toast } = useToast()
   const [isSubmitting, setIsSubmitting] = useState<boolean>(false)
```
Comment (mazipan) [inline src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx]: We don't need the `isSubmitting`, just use `isLoading` from the mutation.

**Comment by mazipan** [comment]
Yes, I prefer to make it smaller.
It will be easier to review also for me.

File: src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx [hunk 2]
```diff
@@ -55,7 +53,7 @@ export default function SettingTelegram({
   existing: NotifChannel[] | null | undefined
 }) {
   const { toast } = useToast()
-  const [isSubmitting, setIsSubmitting] = useState<boolean>(false)
+  const { mutate: mutateUpdateChannelNotif } = usePatchUpdateChannelNotif()
```
Comment (mazipan) [inline src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx]:
```suggestion
  const { mutate: mutateUpdateChannelNotif, isLoading: isLoadingSubmission } = usePatchUpdateChannelNotif()
```

File: src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx [hunk 3]
```diff
@@ -187,7 +180,7 @@ export default function SettingTelegram({
                 <Button
                   type="button"
                   variant="outline"
-                  disabled={!watchUsername || isSubmitting || isLoading}
+                  disabled={!watchUsername || isLoading}
```
Comment (mazipan) [inline src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx]: Revert

File: src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx [hunk 4]
```diff
@@ -200,8 +193,8 @@ export default function SettingTelegram({
         />
 
         <div className="mt-8">
-          <Button type="submit" disabled={isSubmitting || isLoading}>
-            {isSubmitting ? (
+          <Button type="submit" disabled={isLoading}>
```
Comment (mazipan) [inline src/modules/AccountSettings/SettingNotif/SettingTelegram.tsx]: Revert

**Comment by mazipan** [comment]
Conflict ya @madjiebimaa

**Comment by madjiebimaa** [comment]
> Conflict ya @madjiebimaa

Okay mas, I try to resolve this

**Review by mazipan — APPROVED** [review]

**Pull request merged**